src/foam/nanos/notification/ResendNotificationService.java

```diff
@@ -0,0 +1,25 @@
+package foam.nanos.notification;
+import foam.core.ContextAwareSupport;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.nanos.auth.User;
+import foam.nanos.notification.ResendNotificationServiceInterface;
+import foam.nanos.NanoService;
+import foam.nanos.notification.Notification;
+
+public class ResendNotificationService extends ContextAwareSupport implements ResendNotificationServiceInterface, NanoService {
+
+  protected DAO userDAO;
+
+  @Override
+  public void resend(X x, long userId, Notification notification) {
+    User user = (User) userDAO.find(userId);
+    user.doNotify(x, notification);
```

Comment:
@jlhughes in regards to our discussion on slack - notification.resend method where notification can dictate what the resend logic can consist of. This can cover the use cases you mentioned to deactivating previous links and tokens.

Reply:
@zahnno Are you saying the notification can check that it's a resend?  How would this work?